CONTRIBUTING.md

```diff
@@ -1,46 +1,87 @@
 
-### API (Java)
+* Before you get started, look for an issue you would like to work on. Please make sure this issue is not currently assigned to someone else. If there is no existing issue for your idea, please open one.
+* Once you have selected an issue, please discuss your idea for the fix or implementation in the comments for that issue. It's possible that someone else may have a plan for this that the team has discussed, or that there are particular complexities that you should know about before starting the implementation. There are often several ways to fix a problem, and it is essential to find the right approach before spending time on a PR that cannot be merged.
+* Once we've discussed your approach, and everything looks good, we will give you a formal approval to begin development. Please do not start your development work before you get that approval; we don't want you to waste your time on a particular issue if someone is already hard at work on it, or if there's something else in the way!
 
-The initial code uses the default code formatting settings from Spring Tool
-Suite, because they are there. As we develop opinions on what changes should
-be made to those settings, we will add them here.
+We use the `good first issue` label to mark the problems we think will be suitable for new contributors.
 
-In addition, we have a small set of checkstyle rules: these  rules are intended
-to remain few in number, but by the same token they should not be bypassed
-without an extremely compelling reason.
\ No newline at end of file
+### Fork and clone the repository
+
+You will need to fork the main code or documentation repository and clone it to your local machine. See
+[github help page](https://help.github.com/articles/fork-a-repo) for help. 
+
+Create a branch for your work. 
+
+If a you need to base your work on someone elses branch, talk to the branch owner and work something out.  
+
+### Coding your changes
+
+As you code, please make sure you add proper comments, documentation, and unit tests. SimpleReport adheres to strict quality guidelines, regardless of whether the originating engineer is internal or external. Please also ensure that you thoroughly test your changes on your local machine before submission.
+
+
+### Submitting your changes
+
+Once your changes and tests are ready to submit for review:
+
+1. **Test your changes**
+
+    Run the full local test suite to make sure that nothing is broken.
+
+2. **Rebase your changes**
+
+    Update your local repository with the most recent code from the principal repository, and rebase your branch on top of the latest master branch. We prefer your initial changes to be squashed into a single commit. Later, if we ask you to make changes, add the changes as separate commits.  This makes the changes easier to review.  
```

Comment:
Should we use 'main' instead of 'master' for branch name? I think we're trying to move away from using the term master.

Reply:
Great callout on this, as well. I need to clean up the terminology and properly format the branch name...this was a copypasta, and I missed a lot of these smaller things. I really appreciate your attention to detail!